### PR TITLE
use isOutbrainDisabled when evaluating if email-sign up can run

### PR DESCRIPTION
**Pull request by GHaberis**
## What does this change?

- add isOutbrainDisabled to check dispatcher and mediator
- use isOutbrainDisabled check when evaluating whether email-sign up form can run
- use isOutbrainDisabled check in outbrain.js

## What is the value of this and can you measure success?

isOutbrainDisabled is a more appropriate check than thirdPartyTagsDisabled when deciding whether to show email sign-ups.

## Does this affect other platforms - Amp, Apps, etc?

No

## Screenshots

N/A

### CODE REVIEW

File: static/src/javascripts-legacy/projects/commercial/modules/third-party-tags/outbrain.js [hunk 1]
```diff
@@ -108,34 +108,34 @@ define([
...
+                                        return checkMediator.waitForCheck('isUserInNonCompliantAbTest').then(function (userInNonCompliantAbTest) {
+                                            userInNonCompliantAbTest ? module.load('nonCompliant') : module.load();
+                                        });
+                                    }
+                                });
+                            }
+                        });
+                    }
+                });
+            }
```
Comment (SiAdcock) [inline static/src/javascripts-legacy/projects/commercial/modules/third-party-tags/outbrain.js]: What happens if `outbrainDisabled` or `outbrainBlockedByAds` is truthy? No promise is returned, which might lead to exceptions upstream. This (and a fair amount of nesting) can be avoided here by inverting the conditions and instantly returning a resolved promise:
```
if (outbrainDisabled) {
  return Promise.resolve();
}

return loadInstantly() ...

Comment (GHaberis) [inline static/src/javascripts-legacy/projects/commercial/modules/third-party-tags/outbrain.js]: The unit tests are the only place we chain a then to the outbrain.init and the function passed to the then is still executed whether or not we returned a resolved promise or not, however if it's better practise to explicitly return a resolved promise then I can do this.

File: static/src/javascripts-legacy/projects/commercial/modules/third-party-tags/outbrain.js [hunk 2]
```diff
@@ -108,34 +108,39 @@ define([
...
-                                }
-                            });
-                        }
-                    });
-                }
-            });
-        }
-
-        return Promise.resolve(true);
+        return checkMediator.waitForCheck('isOutbrainDisabled').then(function (outbrainDisabled) {
```
Comment (regiskuckaertz) [inline static/src/javascripts-legacy/projects/commercial/modules/third-party-tags/outbrain.js]: That function is becoming unwieldy 😵

File: static/src/javascripts-legacy/projects/common/modules/check-dispatcher.js [hunk 3]
```diff
@@ -43,34 +43,30 @@ define([
...
-                if (thirdPartyTagsDisabled) {
-                    return false;
-                } else {
-                    return trackAdRender('dfp-ad--merchandising-high').then(function(highPriorityAdLoaded) {
-                        return highPriorityAdLoaded;
-                    });
-                }
-            });
+            // if thirdPartyTags false no external ads are loaded
+            if (commercialFeatures.thirdPartyTags) {
```
Comment (regiskuckaertz) [inline static/src/javascripts-legacy/projects/common/modules/check-dispatcher.js]: I'd add `&& commercialFeatures.highMerch` for idempotence.

File: static/src/javascripts-legacy/projects/common/modules/check-dispatcher.js [hunk 4]
```diff
@@ -43,34 +43,30 @@ define([
...
+                return trackAdRender('dfp-ad--merchandising-high').then(function(highPriorityAdLoaded) {
+                    return highPriorityAdLoaded;
```
Comment (regiskuckaertz) [inline static/src/javascripts-legacy/projects/common/modules/check-dispatcher.js]: That `then` callback looks unnecessary.

File: static/src/javascripts-legacy/projects/common/modules/check-dispatcher.js [hunk 5]
```diff
@@ -43,34 +43,30 @@ define([
...
+                });
+            } else {
+                return false;
```
Comment (regiskuckaertz) [inline static/src/javascripts-legacy/projects/common/modules/check-dispatcher.js]: `Promise.resolve(false)`?

File: static/src/javascripts-legacy/projects/commercial/modules/third-party-tags/outbrain.js (hunk 2 above)
Comment (GHaberis) [inline static/src/javascripts-legacy/projects/commercial/modules/third-party-tags/outbrain.js]: Yep I think it needs breaking down, I'm about to check in an improved version.

File: static/src/javascripts-legacy/projects/common/modules/check-dispatcher.js (hunk 3 above)
Comment (GHaberis) [inline static/src/javascripts-legacy/projects/common/modules/check-dispatcher.js]: Cool, will do.

File: static/src/javascripts-legacy/projects/common/modules/check-dispatcher.js (hunk 4 above)
Comment (GHaberis) [inline static/src/javascripts-legacy/projects/common/modules/check-dispatcher.js]: Yep, same for trackAdRender('dfp-ad--merchandising') further down in hasLowPriorityAdLoaded, I'll update both.

File: static/src/javascripts-legacy/projects/common/modules/check-dispatcher.js (hunk 5 above)
Comment (GHaberis) [inline static/src/javascripts-legacy/projects/common/modules/check-dispatcher.js]: This is OK, the return value of this function (in this case - false) is used to resolve a corresponding promise that has been defined in check-mediator, the promise can be resolved with a boolean.

**Comment by GHaberis** [comment]
Hi @SiAdcock @regiskuckaertz, I've made some changes based on reviews and also added some ophan tracking calls which will hopefully provide insight into what leads outbrain to be in the state it's in.

**Comment by regiskuckaertz** [comment]
LGTM 👍  Can you please liaise with @LATaylor-guardian once this is merged? He's working on a new component that will require Outbrain to be in non-compliant mode.

**Review by SiAdcock — APPROVED** [review]
Love it

**Pull request merged**